jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/MassIndexingJobAvailableParameters.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.jsr352.massindexing;
+
+
+/**
+ * @author Yoann Rodiere
+ */
+public final class MassIndexingJobAvailableParameters {
```

Comment:
> we should trust them for the I/O handling. We can create a ticket after having found a failure. For deserialization, I guess we only need to deserialize custom criteria?

We also deserialize booleans and integers, and we don't cleanly validate those right now:

 * if someone puts `"t"` or `"1"` in a boolean property, we'll interpret it as `false` (since we use `Boolean.parseBoolean`). We should probably rather throw an exception, but then it'd be a pain to have the error handling code replicated everywhere we use a boolean property in the codebase.
 * if someone puts "foo" in an integer property, we'll throw a `NumberFormatException` (I think), and this exception won't include detailed context (in particular, the name of the property will be missing).

I'll add a ticket to keep this issue in mind: https://hibernate.atlassian.net/browse/HSEARCH-2632. We may or may not address it, but at least we'll know it's there...